transformation/src/api/pipelineConfigTriggerRequest.ts

```diff
@@ -1,6 +1,6 @@
 export default interface PipelineConfigTriggerRequest {
   datasourceId: number;
 
-  data: object;
+  data: string;
```

Comment:
Why this change? I think it is much cleaner to have the `data` already present as an object and not as stringified JSON.

Reply:
The data attribute is transferred as string (i.e. the json object is enclosed by ''). Changing the type to string was the quickest way of handling the conversion problem. I think you are right that the object type is cleaner, however. I think about if there is a easy solution to achieve that. Otherwise, we could move that to another issue

Reply:
The cleanest solution is changing the adapter behavior such that the enclosing quotes are removed, but this is more complex than it seems (because of persistence in the adapter). I would propose to create another issue for that and use the quick and dirty solution here

Reply:
Ok, I see, But your change will definitely break the REST API endpoint `/api/transformation/trigger` because here the `data` is an object. Maybe we should create two `PipelineConfigTriggerRequest` interfaces, one for amqp and one for rest.

Reply:
And for the `AmqpPipelineConfigTriggerRequest` you can also remove the `dataLocation` property, as it is not used/supported there.

Reply:
I think both of your options are not ideal and not easy to implement. Therefore, I would leave your current PR as it is and send the data as string for now. But I would create a new `AmqpPipelineConfigTriggerRequest` interface that is used inside `pipelineConfigConsumer.ts`:
```ts
export default interface AmqpPipelineConfigTriggerRequest {
  datasourceId: number;
  data: string;
}
```
And the `/api/transformation/trigger` endpoint will still use the old `PipelineConfigTriggerRequest` interface.

I think this solution is also not ideal, because sending the data via RabbitMQ can lead to issue if the data is very large. We should definitely create an issue for this and find a better way of sending the data to the transformation service.

Reply:
Here my two cents on it.

"Make it work. Make it pretty. Make it fast." - that's the order we should go for. 

"Make it work": for a quick fix it is ok to make that change - however make sure to not break other stuff.
Regarding the REST trigger endpoint: is it even used anymore? Maybe we can also remove it if we don't need it anymore... I think it only was used for the migration to amqp.

"Make it pretty": we will have to decide whether to send the data as strings or as object. I have a hunch that the Java annotation magic limits us here (store as string, send as object) or at least makes it more complicated? 

"Make it fast": That's the point with what to send by events. The data itself as fat event, an URL where to fetch or even a reference to a database table as thin event. 
I believe sending fat events is the easiest thing but not the best performing. Making it efficient should be done as soon as we have a use case that brings up that as a bottleneck. IMHO no need to optimize where currently not needed - although it also inches me every time I see it ;)

Reply:
> Regarding the REST trigger endpoint: is it even used anymore? Maybe we can also remove it if we don't need it anymore... I think it only was used for the migration to amqp.
I will create an issue to decide whether we want to keep it for convenience or remove it for simplicity